### ISSUE TITLE
Can't specify a different configuration with --config

### ISSUE THREAD

**Issue by jbonadiman** [AkashRajpurohit/git-sync#2]
### Describe the Bug

I have the use case of several GitHub users to backup, and thought about having separate config files for each one and supplying each with --config.
When I do that, instead of using the one I supplied, git-sync creates a new config file, then it tries to load it and fails (because then it's obviously empty). I also noticed that the home symbol (\~) is not being expanded. In one of my attempts, git-sync created the config file in a folder called "\~" in my home directory.

### Steps to reproduce

Have a valid configuration in ~/.config/git-sync, and try to use it with --config.

### Relevant log output

```shell
> git-sync --config "/home/user/.config/git-sync/config.yml"

2024/07/13 13:06:42 Config file not found, creating a new one...
2024/07/13 13:06:42 Config loaded from: /home/user/.config/git-sync/config.yml
2024/07/13 13:06:42 No username found in config file, please add one.
```

**Comment by AkashRajpurohit** [comment]
Hi @jbonadiman I have pushed a new version 0.1.2 where I have tested using the `~/` syntax for config file path, and it seems to be working for me, can you verify it once on 0.1.2 and report back if this fixes your issue?

<img width="602" alt="image" src="https://github.com/user-attachments/assets/3b8e2a1c-ca0e-4361-9de3-57c57951289d">

**Comment by jbonadiman** [comment]
> Hi @jbonadiman I have pushed a new version 0.1.2 where I have tested using the `~/` syntax for config file path, and it seems to be working for me, can you verify it once on 0.1.2 and report back if this fixes your issue?
> 
> <img alt="image" width="602" src="https://private-user-images.githubusercontent.com/30044630/348477843-3b8e2a1c-ca0e-4361-9de3-57c57951289d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4ODU4ODMsIm5iZiI6MTcyMDg4NTU4MywicGF0aCI6Ii8zMDA0NDYzMC8zNDg0Nzc4NDMtM2I4ZTJhMWMtY2EwZS00MzYxLTlkZTMtNTdjNTc5NTEyODlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEzVDE1NDYyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhYmE5ZDYxMGZmMjJiMDhhNTM4NjJkMWVlNmZlNTY5OTcwOWYwMzgyNTMyYWQ2YjM3YzhmMWUzYjVmMmM1OGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YArNhQHHaLF2CMJvLOMTs-JvygRPpckVsHJ3QSwcn-Y">

I see that the `~/` syntax now works, but somehow every time I point to a config file, instead of using it, git-sync says it didn't find it and resets it, like so:
```
> git-sync --config user.yaml
2024/07/13 16:50:06 Config file not found, creating a new one...
2024/07/13 16:50:06 Config loaded from: user.yaml
2024/07/13 16:50:06 No username found in config file, please add one.
```
And then I end up with a template configuration, like so:
```
backup_dir: /home/user/git-backups
include_all_repos: true
include_forks: false
repos: []
token: ""
username: ""
```

**Comment by AkashRajpurohit** [comment]
That's weird because I was able to pass a custom path to the config file and it worked fine (the image attached above)

Can you confirm these things

1. From your example was the user.yaml file already present with the config schema before hand? Or did you just point it to a file name which was not present and git-sync created that file for you?
2. Once you ran it and it said there is no username, did you edit that file with the required values like username and token and ran it again with the same file path and then it still failed? This should not happen ideally

**Comment by jbonadiman** [comment]
> That's weird because I was able to pass a custom path to the config file and it worked fine (the image attached above)
> 
> Can you confirm these things
> 
> 1. From your example was the user.yaml file already present with the config schema before hand? Or did you just point it to a file name which was not present and git-sync created that file for you?
> 2. Once you ran it and it said there is no username, did you edit that file with the required values like username and token and ran it again with the same file path and then it still failed? This should not happen ideally

I'm sorry, I wasn't very clear:
1. Yes, I had a valid configuration file beforehand.
2. Yes, the file previously contained a valid and tested configuration.

git-sync first action was to create the configuration file, effectively rewriting the one I had. Then, it tries to use it and of course, the username is empty and that's where that log message comes from.

**Comment by AkashRajpurohit** [comment]
> git-sync first action was to create the configuration file, effectively rewriting the one I had

Let me try if I can reproduce this, from the yesterday's testing seems like it is working fine for me. When I pointed it to a file `~/.config/tmp/config.yaml` it did pick the file correctly (without overriding anything) and then ran accordingly.

**Comment by AkashRajpurohit** [comment]
Pushed a fix 0.1.3 which should fix it, I have tested the scenario that you have mentioned as well. Once you have verified it, we can close this ticket, thanks for opening it up!

**Comment by jbonadiman** [comment]
It's working beautifully now, thanks for the quick replies and the awesome work!
